sample_consensus/include/pcl/sample_consensus/impl/sac_model_line.hpp

```diff
@@ -221,7 +221,12 @@ pcl::SampleConsensusModelLine<PointT>::optimizeModelCoefficients (
 
   // Compute the 3x3 covariance matrix
   Eigen::Vector4f centroid;
-  compute3DCentroid (*input_, inliers, centroid);
+  if (0 == compute3DCentroid (*input_, inliers, centroid))
+  {
+    PCL_DEBUG ("[pcl::SampleConsensusModelLine::optimizeModelCoefficients] compute3DCentroid failed (returned 0), possibly because there are no valid inliers.\n");
```

Comment:
The amount of inliers is checked to be minimum sample_size_ in line 213? But if this is 0..?

Reply:
The return value can be smaller than `inliers.size()` if not all inliers are finite

Reply:
Ahh, true.